### ISSUE TITLE
OPC node value update event

### ISSUE THREAD

**Issue by fch96733357** [nauful/LibUA#85]
Hi, 
      I am using LibUA .net library to integrate with my database system, I am able to create nodes from database, and update nodes value from my application using socket; I just wonder how to create write request event for my nodes (read&write), when client is updating (set new value) the node's value, I am able to capture the new value so that I can pass the new value to my application. Something like OnValuechanged(node mnode,int newvalue).

Thanks and best regards
Feng Chonghai
Fch96733357@gmail.com

**Comment by nauful** [comment]
Try implementing `public virtual UInt32[] HandleWriteRequest(object session, WriteValue[] writeValues)` in your server `Application`.  This will notify you of any values written over UA to your database.

**Comment by fch96733357** [comment]
Nauful, Thank you very much for quick response.

      Yes, I did try to implement HandleWriteRequest virtual function, however, when I tried to write new value via OPC UA client, debug windows shown below
The new value does not in writeValues[], is it this correct?

Best Regards
Feng



Sent from Mail for Windows

**Comment by nauful** [comment]
Hello Feng,

Could you please paste the screenshot in Github? I don't see any image if you replied via email.

**Comment by fch96733357** [comment]
Hi Nauful,

     Thanks for your great OPC lib, it solved my problem. Thanks a lot

**Comment by fch96733357** [comment]
Close

**Comment by fch96733357** [comment]
Thanks Nauful, I have managed to get it work, because I did not initialize value.

Best Regards
Feng

Sent from Mail for Windows

From: Feng ChongHai
Sent: 2022年7月12日 8:19
To: nauful/LibUA; nauful/LibUA
Cc: Author
Subject: RE: [nauful/LibUA] OPC node value update event (Issue #85)

Nauful, Thank you very much for quick response.

      Yes, I did try to implement HandleWriteRequest virtual function, however, when I tried to write new value via OPC UA client, debug windows shown below
The new value does not in writeValues[], is it this correct?

Best Regards
Feng



Sent from Mail for Windows
